sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -155,15 +154,20 @@ export class ServiceBusClient {
     constructor(connectionString: string, options?: ServiceBusClientOptions);
     constructor(hostName: string, tokenCredential: TokenCredential, options?: ServiceBusClientOptions);
     close(): Promise<void>;
+    getDeadLetterReceiver(queueName: string, receiveMode: "peekLock"): Receiver<ReceivedMessageWithLock>;
```

Comment:
`getDeadLetterReceiver`, is this necessary?
Users might think this to be something completely different where as this is essentially the exact same receiver.
Instead, we could just provide the path-formatter which would effectively let users know more about the service behavior.

Are we sure we want to do it this way?

Reply:
There are a couple of reasons I like this:

1. It advertises that dead lettering exists as a feature. 

  If you're going down the "optimal" route with service bus, where you are a good distributed-citizen, peekLock is the only way to do messaging where you're resilient against message loss.

  DeadLettering is a big part of that - you can ignore your dead letter queue but you shouldn't. So letting people know it exists is important.

2. It makes it obvious how to instantiate the receiver properly.

  Creating the dead letter receiver is a bit odd - it's _always_ a queue (even if you chose a subscription) and it doesn't support sessions. Both of those are not necessarily known right off the bat so the user could go down the wrong path for no good reason.

  If we go down the EntityPathFormatter route then the user has to piece all of that together themselves. 

So in short - I do like it this way. I do hate to find myself agreeing with @ramya-rao-a  but on occasion....it just makes sense. :)